pkg/nsx/services/staticroute/staticroute.go

```diff
@@ -0,0 +1,138 @@
+package staticroute
+
+import (
+	"sync"
+
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"k8s.io/client-go/tools/cache"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	common "github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+type StaticRouteService struct {
+	common.Service
+	StaticRouteStore *StaticRouteStore
+}
+
+var (
+	log                     = logf.Log.WithName("service").WithName("staticroute")
+	resourceTypeStaticRoute = "StaticRoutes"
+	resourceType            = "resource_type"
+	String                  = common.String
+)
+
+// InitializeStaticRoute sync NSX resources
+func InitializeStaticRoute(commonService common.Service) (*StaticRouteService, error) {
+	wg := sync.WaitGroup{}
+	wgDone := make(chan bool)
+	fatalErrors := make(chan error)
+
+	wg.Add(1)
+	staticRouteService := &StaticRouteService{Service: commonService}
+	staticRouteStore := &StaticRouteStore{}
+	staticRouteStore.Indexer = cache.NewIndexer(keyFunc, cache.Indexers{
+		common.TagScopeStaticRouteCRUID: indexFunc,
+	})
+	staticRouteStore.BindingType = model.StaticRoutesBindingType()
+	staticRouteService.StaticRouteStore = staticRouteStore
+	staticRouteService.NSXConfig = commonService.NSXConfig
+
+	go staticRouteService.InitializeResourceStore(&wg, fatalErrors, resourceTypeStaticRoute, staticRouteService.StaticRouteStore)
+
+	go func() {
+		wg.Wait()
+		close(wgDone)
+	}()
+
+	select {
+	case <-wgDone:
+		break
+	case err := <-fatalErrors:
+		close(fatalErrors)
+		return staticRouteService, err
+	}
+
+	return staticRouteService, nil
+}
+
+func (service *StaticRouteService) CreateorUpdateStaticRoute(obj *v1alpha1.StaticRoute) error {
+	nsxStaticRoute, err := service.buildStaticRoute(obj)
+	if err != nil {
+		return err
+	}
+
+	existingStaticRoute := service.StaticRouteStore.GetByKey(*nsxStaticRoute.Id)
+	if existingStaticRoute != nil && service.compareStaticRoute(existingStaticRoute, nsxStaticRoute) {
+		return nil
+	}
+	// TODO, update the value from 'default' to actual value
+	err = service.patch("default", "default", "default", nsxStaticRoute)
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
1. also need to add staticroute to store
2. need to get nsx staticroute path and update it to StaticRoute obj